products/iam/api.yaml

```diff
@@ -195,4 +198,236 @@ objects:
       - !ruby/object:Api::Type::Boolean
         name: 'deleted'
         description: The current deleted state of the role
-        output: true
\ No newline at end of file
+        output: true
+  - !ruby/object:Api::Resource
+    name: 'WorkloadIdentityPool'
+    min_version: beta
+    base_url: projects/{{project}}/locations/global/workloadIdentityPools
+    create_url: projects/{{project}}/locations/global/workloadIdentityPools?workloadIdentityPoolId={{id}}
```

Comment:
The `{{id}}` field will also need to appear as a field under `properties` (or `parameters`, but  being in that list instead doesn't actually do anything).

You can then add `url_param_only: true` to indicate that it's only used in the URL. See https://github.com/GoogleCloudPlatform/magic-modules/blob/6c2be6af37835602ec918fa7a0b07e4698b9892c/products/servicedirectory/api.yaml#L50-L57 for an example.

I'd also recommend avoiding the `id` name, just because Terraform treats `id` as a reserved field name. `workloadIdentityPoolId` from https://cloud.google.com/iam/docs/reference/rest/v1beta/projects.locations.workloadIdentityPools/create seems appropriate.